src/NuGet.Core/NuGet.Versioning/SemanticVersionBase.cs

```diff
@@ -178,9 +182,12 @@ public virtual int CompareTo(SemanticVersion other, VersionComparison versionCom
             return Compare(version1, version2) >= 0;
         }
 
-        private static int Compare(SemanticVersion version1, SemanticVersion version2)
+        private static int Compare(SemanticVersion? version1, SemanticVersion? version2)
         {
+#pragma warning disable CS8604 // Possible null reference argument.
```

Comment:
Should permanent suppressions be in GlobalSuppressions instead?
Hide them, pretend it doesn't exist :D

Reply:
For these one-line methods, GlobalSuppressions could be used. Although, if anyone refactors the method in the future, then they won't have null checking protection from the compiler. A problem with GlobalSuppressions.cs is that it applies to entire methods, so I definitely wouldn't want to use it for multi-line method implementations, but that's the not case for this specific example. Personally, I prefer the inline suppressions. I don't like the GlobalSuppressions file, even when I want a suppression to apply to the entire method body.

Reply:
I'll resolve this. There's so many other changes not worth focusing on this. 
Alternatively we can suppress on top of the method too.